server/app/services/export/CsvExporter.java

```diff
@@ -22,59 +21,39 @@
  * {@link Path} indexing into an applicant's data, and CsvExporter takes the path and reads the
  * answer from {@link ReadOnlyApplicantProgramService} if present.
  */
-public class CsvExporter {
-  public static final CSVFormat DEFAULT_CSV_FORMAT =
-      CSVFormat.DEFAULT.builder().setHeader().setSkipHeaderRecord(true).build();
-
+public final class CsvExporter implements AutoCloseable {
   private final String EMPTY_VALUE = "";
 
-  private boolean wroteHeaders;
-  private ImmutableList<Column> columns;
-  private Optional<String> secret;
-
-  public CsvExporter(List<Column> columns) {
-    this.wroteHeaders = false;
-    this.columns = ImmutableList.copyOf(columns);
-    this.secret = Optional.empty();
-  }
+  private final ImmutableList<Column> columns;
+  private final Optional<String> secret;
+  private final CSVPrinter printer;
 
   /** Provide a secret if you will need to use OPAQUE_ID type columns. */
-  public CsvExporter(ImmutableList<Column> columns, String secret) {
-    this(columns);
+  public CsvExporter(ImmutableList<Column> columns, String secret, Writer writer)
+      throws IOException {
+    this.columns = columns;
     this.secret = Optional.of(secret);
-  }
 
-  private void writeHeadersOnFirstExport(CSVPrinter printer) throws IOException {
-    if (!wroteHeaders) {
-      for (Column column : columns) {
-        printer.print(column.header());
-      }
-      printer.println();
-      wroteHeaders = true;
-    }
-  }
-
-  protected ImmutableList<Column> getColumns() {
-    return columns;
+    CSVFormat format =
+        CSVFormat.DEFAULT
+            .builder()
+            .setHeader(columns.stream().map(Column::header).toArray(String[]::new))
+            .build();
+    this.printer = new CSVPrinter(writer, format);
   }
 
   /**
-   * The CSV exporter will write the headers on first call to services.export(). It does not store
-   * the writer between calls. Since it is intended for many applications, this function is intended
-   * to be called several times.
+   * The CSV exporter does not store the writer between calls. Since it is intended for many
+   * applications, this function is intended to be called several times.
    */
```

Comment:
suggestion (blocking): the comment for this method beginning with "The CSV exporter does not store the writer between calls." is a little strange now that the method doesn't take a writer. Please reword or remove